src/utils_dpdk.c

```diff
@@ -705,10 +727,12 @@ static void dpdk_helper_check_pipe(dpdk_helper_ctx_t *phc) {
     }
   }
   while (data_avail) {
-    int nbytes = read(phc->pipes[0], buf, sizeof(buf));
+    int nbytes = read(phc->pipes[0], buf, (sizeof(buf) - 1));
+    DEBUG("%s:dpdk_helper_check_pipe: read nbytes=%d", phc->shm_name, nbytes);
     if (nbytes <= 0)
       break;
-    sstrncpy(out, buf, nbytes);
+    buf[nbytes] = '\n';
+    sstrncpy(out, buf, (nbytes + 1));
```

Comment:
Use `sizeof(out)` as the third parameter (after making sure that `buf` is nil-terminated, see above).

Reply:
sstrncpy does not check nil in string, so it will copy complete buffer. In our case only real data are copied.

Reply:
`sstrncpy()` expects a **nil-terminated** string as second argument and copies it into the memory pointed to by the first argument. It will, however, not copy more than the third argument.

While your variant also works, it is uncommon and therefore confusing. Please use `sizeof(out)` instead of `nbytes + 1`.

Reply:
I've changed that as you requested. Nevertheless if I understood documentation right then it is less efficient than original one. It always write complete buffer even if read nbytes  is only few chars. Rest of buffer is filled with zeros.